__snapshots__/website-create.test.js

```diff
@@ -74,6 +93,13 @@ Created website: 9caf90c31c4449f195a1a40acc979cf0
 
 `
 
+exports['website create using active profile with name, app name, historyApiRouting and errorPage should fail 1'] = `
```

Comment:
Is this CLI logic? If yes, then it should be moved to the server. If not, we do not need a test for it, I think.

Reply:
It's both server-side and client-side logic.
When you type `website create` the CLI creates a website as well as a site environment and this is isn't something known by the user. Therefore, it won't be nice if site creation succeeds but siteEnv creation fails. So, there's validation in the CLI for `historyApiRouting` and `errorPage` (which are siteEnv fields).

Reply:
What I meant is that here you are testing that if you supply historyApiRouting=true and non-empty error page the command will fail. As far as I understand, this is not CLI logic (should not be). The CLI only supplies arguments to the server. This should be server logic and should be tested in the server.

You should have a test case where creation fails in order to ensure proper CLI error handling, but not have test cases for each case where you know the server will fail.

Reply:
This is server-side logic.
But there is also validation in the CLI for the same thing - if `historyApiRouting` is `true`, then `errorPage` cannot be supplied. Why? Because users don't know/see that we actually have environments in the websites. From a user perspective `website create` is an atomic operation and only one entity is created. In reality, we create a website and then an environment. If env creation fails, things get a bit weird. To minimize the risk of env creation failing, we have client-side validation for routing + errorPage. Hence, the test in question.

Reply:
OK, understood.